controllers/src/main/java/com/epam/esm/controllers/RestResponseEntityExceptionHandler.java

```diff
@@ -0,0 +1,59 @@
+package com.epam.esm.controllers;
+
+import com.epam.esm.services.exceptions.ErrorResponse;
+import com.epam.esm.services.exceptions.ResourceNotFoundException;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.servlet.NoHandlerFoundException;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
+import java.util.Locale;
+
+@ControllerAdvice
+public class RestResponseEntityExceptionHandler extends ResponseEntityExceptionHandler {
+
+    private static final Logger LOGGER = LogManager.getLogger(RestResponseEntityExceptionHandler.class);
+    private static final String INTERNAL_ERROR = "message.exception.internal";
+    private static final String PAGE_NOT_FOUND_ERROR = "message.exception.page.notfound";
+
+    @ExceptionHandler(ResourceNotFoundException.class)
+    protected final ResponseEntity<ErrorResponse> handleResourceNotFoundException(ResourceNotFoundException exception,
+                                                                                  WebRequest request) {
+        Locale locale = request.getLocale();
+        exception.setLocale(locale);
+        String errorMessage = exception.getLocalizedMessage() + exception.getResource();
+        int errorCode = ResourceNotFoundException.getErrorCode();
+        LOGGER.debug("errorMessage = " + errorMessage);
+        ErrorResponse error = new ErrorResponse(errorMessage, errorCode);
+        return new ResponseEntity<>(error, ResourceNotFoundException.getHttpStatus());
+    }
+
+    @Override
+    protected ResponseEntity<Object> handleNoHandlerFoundException(NoHandlerFoundException ex, HttpHeaders headers,
+                                                                   HttpStatus status, WebRequest request) {
+        Locale locale = request.getLocale();
+        String errorMessage = ErrorResponse.getMessageForLocale(PAGE_NOT_FOUND_ERROR, locale);
+        int errorCode = 40402;
+        ErrorResponse error = new ErrorResponse(errorMessage, errorCode);
+        return new ResponseEntity<>(error, HttpStatus.NOT_FOUND);
+    }
+
+    @Override
+    @ExceptionHandler(Exception.class)
+    protected ResponseEntity<Object> handleExceptionInternal(Exception exception, Object body, HttpHeaders headers,
+                                                             HttpStatus status, WebRequest request) {
+        Locale locale = request.getLocale();
+        String errorMessage = ErrorResponse.getMessageForLocale(INTERNAL_ERROR, locale);
+        int errorCode = 50001;
```

Comment:
Not sure.., Can I see somewhere all the possible error codes of the application?